tombstone_gc.cc

```diff
@@ -285,6 +285,21 @@ std::map<sstring, sstring> get_default_tombstone_gc_mode(data_dictionary::databa
     return get_default_tombstone_gc_mode(ks.get_replication_strategy(), tm);
 }
 
+void validate_tombstone_gc_options(const tombstone_gc_options* options, const gms::feature_service& feature_service,
+        const locator::abstract_replication_strategy& rs, const locator::token_metadata& tm)
+{
+    if (!options) {
+        return;
+    }
+    if (!feature_service.tombstone_gc_options) {
+        throw exceptions::configuration_exception("tombstone_gc option not supported by the cluster");
+    }
+
+    if (options->mode() == tombstone_gc_mode::repair && !needs_repair_before_gc(rs, tm)) {
+        throw exceptions::configuration_exception("tombstone_gc option with mode = repair not supported for table with RF one or local replication strategy");
+    }
+}
+
```

Comment:
>  tombstone_gc: Add overload of validate_tombstone_gc_options
> 
> We're adding it for consistency with `get_default_tombstone_gc_mode`.
> It's not used for anything yet.

The commit message is confusing. The new overload _is_ being used. This commit moves the "meat" of the existing `validate_tombstone_gc_options` function into an overload which does not need to perform as many checks / call as many getters, but the original overload does use the new one.

I guess that you meant that it is not used by any outside code yet. Because of that I am not sure that it makes sense to expose this function at the moment.